webcurator-ui/src/utils/rest.api.ts

```diff
@@ -189,77 +193,112 @@ export function useFetch() {
 
         let reqPath;
         if (path.startsWith('/')) {
-          reqPath = RootContextPath + path;
+          reqPath = ApiContextPath + path;
         } else {
-          reqPath = RootContextPath + '/' + path;
+          reqPath = ApiContextPath + '/' + path;
         }
 
-        const rsp = await fetch(reqPath, reqOptions).catch((err: any) => {
-          toast.removeAllGroups();
-          toast.add({
-            severity: 'warn',
-            summary: 'Warning! ',
-            detail: err.message,
-            life: ToastLife
-          });
-        });
-
-        //Exception has happened
-        if (!rsp) {
-          // await sleep(RetryDelay);
-          // continue;
-          break;
+        let rsp;
+        try {
+          rsp = await fetch(reqPath, reqOptions);
+        } catch (err: any) {
+          retriedTimes.value++;
+          if (retriedTimes.value >= MaxRetryTimes) {
+            confirm.error(`Failed to [${methodValue}] ${reqPath}: ${err.message}`);
+            break;
+          } else {
+            confirm.warning(`Failed to [${methodValue}] ${reqPath}: ${err.message}.  Will retry in ${RetryDelay / 1000} seconds.`);
+            await sleep(RetryDelay);
+            continue;
+          }
         }
 
-        //Need authentication, forward to login page
-        if (rsp.status === 401) {
+        if (!rsp) {
+          //Exception has happened
+          confirm.error('Failed to [' + methodValue + '] ' + reqPath);
+          break;
+        } else if (rsp.status === 502 || rsp.status === 504) {
+          //Upstream error
+          const err = await extractErrorMessageFromResponse(rsp);
+          retriedTimes.value++;
+          if (retriedTimes.value >= MaxRetryTimes) {
+            confirm.error(`Failed to [${methodValue}] ${reqPath}: ${err}`);
```

Comment:
Errors in this format are useful for the console or logs, but not very readable for non-technical users. I think they should be more simple, e.g. "Server error, unable to save target". Perhaps once the api returns error messages, we could replace these